tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java

```diff
@@ -64,13 +72,45 @@ public TsFileSequenceReader(String file) throws IOException {
    * @param loadMetadataSize -load meta data size
    */
   public TsFileSequenceReader(String file, boolean loadMetadataSize) throws IOException {
+    this(file, loadMetadataSize, false);
+  }
+
+  /**
+   * construct function for TsFileSequenceReader.
+   *
+   * @param file -given file name
+   * @param loadMetadataSize -load meta data size
+   * @param autoRepair if true it tries to automatically repair a damaged file (by changing it!)
+   *                   before reading
+   */
+  public TsFileSequenceReader(String file, boolean loadMetadataSize, boolean autoRepair) throws IOException {
     this.file = file;
-    tsFileInput = new DefaultTsFileInput(Paths.get(file));
+    final Path path = Paths.get(file);
+    tsFileInput = new DefaultTsFileInput(path);
+    if (autoRepair) {
+      checkAndRepair(file, path);
+    }
     if (loadMetadataSize) {
       loadMetadataSize();
     }
   }
 
+  /**
+   * Checks if the file is incomplete, and if so, tries to repair it.
+   */
+  private void checkAndRepair(String file, Path path) throws IOException {
+    // Check if file is damaged
+    if (!TSFileConfig.MAGIC_STRING.equals(readTailMagic())) {
+      // Try to close it
+      LOGGER.info(String.format("File %s has no correct tail magic, try to repair...",
```

Comment:
you can use `LOGGER.info("File {} has no correct tail magic, try to repair...", path.toAbsolutePath()),` directly

Reply:
You are absolutely right @MyXOF . Fixed it.